### DISCUSSION TITLE
Enable syncing /addon_configs folder

### DISCUSSION THREAD

**Discussion by donverse** [lavinir/hassio-onedrive-backup#157] [Ideas]
Hi - first of all: many thanks for your reliable and very easy-to-use add-on!

Here my request (or ask, if you will!): I have installed Paperless-ngx on my HA system. This add-on is installed to the addon_configs folder rather than to addons. Can you please enable the addon_configs folder to be synced via your 'file sync' functionality? Currently, this folder is not available. It would be perfect to be able to sync the raw paperless data (i.e. the stored pdf files) as I don't need to back them up every single time - they can be treated similarly to media files.

**Comment by lavinir** [comment]
Regarding adding additional sync folders - at the time I added this those were the only folders allowed to be mapped to an addon container, now I see more exist but it's addon_config not addon_configs according to the documentation. I will add support for the new folders.

Out of curiosity, I am using Paperless-ngx as well but why would you run this inside Home Assistant ? What integration exists there ?

**Reply by donverse** [reply]
Thanks! The folder name indeed is addon_configs (with the added 's'; [source](https://developers.home-assistant.io/blog/2023/11/06/public-addon-config/) from the developer documentation.
I am running paperless ngx via [this addon](https://github.com/alexbelgium/hassio-addons/tree/master/paperless_ngx) from Alex Belgium; mainly because I just wanted to try it out easily with my existing RPi4 without having to buy anything. I have currently added a few dozen - 100 documents and while the processor usage maxes out during ocr, the performance is not terrible at all.

**Reply by lavinir** [reply]
seems a strange use case to use Home Assistant to host unrelated docker containers but alright :)
Anyway looking more into this I don't think it will solve your problem. This full contents of the addon_configs folder is not available for the addons - you can opt in to this and you only receive a mapping to your own dedicated addon folder within so I wouldn't be able to read files located in the addon_config subfolder for Paperless..
However, in terms of backup, that folder should be part of your hass backup for the Paperless addon so the OneDrive backup will still back that up to OneDrive

**Reply by donverse** [reply]
Yes - I confirm paperless as addon (with all contents) is backed up - strangely only after I deleted your addon and set it up again; before this it was missing a few addons. This is a bit waste of space of course since all the PDFs are backed up each day, but not a major concern.
Just thinking out loud - there are a few addons which have full access to addon_configs (namely Samba, SSH, Studio Code Server - [link](https://developers.home-assistant.io/blog/2023/11/06/public-addon-config/#new-addon_configs-folder)), could it be an option for you to enable this all_addon_configs:rw setting as an optional flag for your addon? Perhaps I am thinking to simple, but then users could decide if they'd need/want this or not. Thanks anyway for your replies - much appreciated.

**Reply by lavinir** [reply]
Regarding the missing addons. A restart of the addon would have fixed this but I took a look and indeed I am not updating the existing addon list before performing the backup so if you are creating 'partial' backups it would only back up the addons it found when the addon was loaded. I will open a bug for this.

For the all_addon_configs:rw - I missed this yesterday I will take another look at this and if it makes sense I will add this scope.

**Reply by donverse** [reply]
Yes indeed, i was performing a partial backup (excluding media folder which makes it partial I guess). Thanks for acknowledging and fixing 👍
Restarting addon and/or home assistant before did not resolve it earlier though.

Looking forward to your decision on the rw setting, it's not crucial but would be great to have, especially if more add-ons start utilizing the addon_configs folder. Keep the great work up!

**Comment by donverse** [comment]
THANK you for implementing, it does exactly what I was looking for. MUCH appreciated!
